services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/GenericS3TransferManager.java

```diff
@@ -136,11 +137,15 @@ public Upload upload(UploadRequest uploadRequest) {
         requestBody = progressUpdater.wrapRequestBody(requestBody);
         progressUpdater.registerCompletion(returnFuture);
 
+        Consumer<AwsRequestOverrideConfiguration.Builder> attachProgressListener =
+            b -> b.putExecutionAttribute(JAVA_PROGRESS_LISTENER, progressUpdater.javaProgressListener());
+        PutObjectRequest putObjectRequest = attachSdkAttribute(uploadRequest.putObjectRequest(), attachProgressListener);
```

Comment:
Don't we need to add branching here as well for clients with multipart disabled?